acceptance/tests/base/dsl/helpers/host_helpers_test.rb

```diff
@@ -0,0 +1,1377 @@
+require "fileutils"
+
+# NOTE: Currently scp failures throw a Net::SCP::Error exception in SSH connection
+# close, which ends up not being caught properly, and which ultimately results
+# in a RuntimeError. The SSH Connection is left in an unusable state and all
+# later remote commands will hang indefinitely.
+#
+# TODO: fix via: https://tickets.puppetlabs.com/browse/BKR-464
+def test_scp_error_on_close?
+  !!ENV["TEST_SCP_ERROR_ON_CLOSE"]
+end
+
+# NOTE: currently there is an issue with the tmpdir_on helper on cygwin
+#       platforms:  the `chown` command always fails with an error about not
+#       recognizing the Administrator:Administrator user/group.  Until this is
+#       fixed, we add this shim that delegates to a non-`chown`-executing version
+#       for the purposes of our test setup.
+#
+# TODO: fix via: https://tickets.puppetlabs.com/browse/BKR-496
+def tmpdir_on(hosts, path_prefix = '', user=nil)
+  return create_tmpdir_on(hosts, path_prefix, user) unless Array(hosts).first.is_cygwin?
```

Comment:
Not a big fan of this - if you are using block_on that means that hosts could be just a host name, which won't have an is_cygwin? method.

Look into confine_block in the beaker dsl.